2018/001-containers-consumption-from-registry.md

```diff
@@ -72,10 +72,70 @@ by the `container-feeder` are always tagged in the host with the RPM release
+## Suggested approach
+
+The suggested approach takes into consideration the following requirements:
+
+* No reboot is required to update any contianer within the admin node and
+  within the cluster.
+* No OBS tricks are done at build time to chain packages and image builds
+  (hacks like containment-rpm-docker)
+* Allow/Require human interaction to trigger image updates (coding something
+  like an `apply updates` button should be possible)
+ 
+Having constant manifest with mutable image tags and make use of the
+`imagePullPolicy="Always"` kubernetes flag does not fulfill our needs.
+First, it does not avoid the potential issues derived of adding nodes
+dynamically as they could silently start running different images given
+the same image referece. Second, in that context providing some kind of
+human gate to apply image updates over the cluster is not simple. Thus
+using mutable tags and immutable kubernetes manifests is not considered
+for this proposal.
+
+An obvious consequence of that are dynamic kuberentes manifests that change
+over time as image updates are applied. This can be splitted in some
+different steps:
+
+1- Updates discovery (discover new images in registry)
+2- Manifest update (modify and apply manifests)
+3- Clean unused images after update
+
+In this case a human gate to trigger updates could be easily placed
+between step 1 and 2.
+
+Given that we assume mutable kubernetes manifests the suggestion is
+to make them as explicit as possible, thus use immutable tags. Meaning each
+image reference in a manifest points to a unique image. Our current tagging
+strategy defined in [TEXTFORTHELINK](link to the tagging strategy) currently
+includes a unique reference for each image builds. Moreover, the tags are
+enforced to follow a versioning schema as it is done with packages, so they
+are comparable. A procedure to identify released updates could be similar to:
+
+1- Get a list of the images referenced in manifest
+2- For each image check the higest tag available in the registry
+   (warning: this must work for registry.suse.com and for any other
+   public registry)
+3- Compare the higest tag with the one included into the manifest
+4- Raise update flag for manifest X
+
+This procedure is quite similar to what a package manger does to check updates
+from repositories. At the moment there is no tool available to perform this
+task. This is likely to require coding a new tool from scratch.
+
+(Thoughts about containerizing the service)
+
+Another minor challenge of this approach is how to set the initial manifest.
+In that case there could be two different options:
+
+1- Ship manifests as templates, thus assume the manifest requires always
+   an initial update before it can actually be used.
+
+2- Ship manifests using some mutable tag (e.g latest). Then the manifest
+   is usable from the very beginning, but first update should be considered
+   an strict requirement (probably automated without any user interaction)
+
 
```

Comment:
This approach looks simple enough to be doable. However, I would like to hear @ereslibre and @flavio opinion on this.

Then, if we like this approach, we better ask SCC team to make sure registry.suse.com works as expected and the confirmation that this is going to be stable.